python/triton/language/semantic.py

```diff
@@ -1228,15 +1228,28 @@ def is_hip():
         raise ImportError("Triton requires PyTorch to be installed")
     return torch.version.hip is not None
 
-def gpu_has_mfma() -> bool:
+
+def gpu_matrix_core_version() -> int:
+    """ Determine matrix core type available on current GPU.
+
+        0 means no tensor cores are available
+        1 corresponds to MFMA in CDNA 1 architecture
+        2 corresponds to MFMA in CDNA 2 architecture
```

Comment:
Does gfx94x refer to MI300? If so, we also use 2 for CDNA 3.

Reply:
And are we using this `gpu_matrix_core_version` as the counterpart of `computeCapability`?

Reply:
> Does gfx94x refer to MI300? If so, we also use 2 for CDNA 3.

For now, yes. If there are some changes to MFMA instruction, we can add another layer later.
I'll add CDNA3 in this comment

Reply:
> And are we using this gpu_matrix_core_version as the counterpart of computeCapability?

yes, this is kind of strange, but I can not think of better way for now.

Maybe rename `computeCapability` into something like `MatrixCoreVersion`/`tensorCoreVersion`?

Reply:
another idea I have is to separate nvidia and amd `accelerateMatmul` passes.
There are almost no common code so this should be a trivial change.

I am also looking in upstream idea of nvidia specific dialect separation.
Maybe we should wait and refactor mfma code according to upstream.